Prevent multiple contributions to Panache's "entityToPersistenceUnit"  from overriding each other

yrodiere opened the pull request:
This should fix the seemingly random "Entity not found" when using Kotlin.

* Fixes #50955

yrodiere committed:
Prevent multiple contributions to Panache's "entityToPersistenceUnit" from overriding each other

This should fix the seemingly random "Entity not found" when using Kotlin.

yrodiere commented:
Thanks for the review @lucamolteni . @FroMage, mind adding your stamp here so I can merge? :)

FroMage commented on extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/AbstractJpaOperations.java:
```diff
@@ -25,12 +25,22 @@
+    // Putting synchronized here because fields involved were marked as volatile initially,
+    // so I expect recorders can be called concurrently?
+    public static synchronized void addEntityTypesToPersistenceUnit(Map<String, String> map, boolean incomplete) {
+        // Note: this may be called multiple times if an app uses both Java and Kotlin.
```
Well, whatever.

yrodiere commented:
Thank you @FroMage!

rsvoboda commented:
@yrodiere why is this proposed for 3.27?

https://github.com/quarkusio/quarkus/issues/50955 says that the app builds/tests fine with Quarkus 3.28.4, problem was in 3.29.z.

yrodiere commented:
@rsvoboda the problem I fixed was probably present in 3.27 given the code involved, which is why I proposed to backport to 3.27.
But it turned out the problem I fixed was not actually #50955 -- which was fixed in https://github.com/quarkusio/quarkus/pull/51213 instead.

So... we can probably skip backporting this, especially if it conflicts.
